PWGEM/PhotonMeson/TableProducer/createEMReducedMCEvent.cxx

```diff
@@ -110,22 +111,22 @@ struct createEMReducedMCEvent {
 
       for (auto& mctrack : groupedMcTracks) {
         if (mctrack.pt() < 1e-2 || abs(mctrack.y()) > 1.5 || abs(mctrack.vz()) > 250 || sqrt(pow(mctrack.vx(), 2) + pow(mctrack.vy(), 2)) > 500) { // if pT < 10 MeV/c, don't store. Anyway, we don't care such low pT particles.
+                                                                                                                                                   // if (mctrack.pt() < 1e-2 || abs(mctrack.vz()) > 250 || sqrt(pow(mctrack.vx(), 2) + pow(mctrack.vy(), 2)) > 500) { // if pT < 10 MeV/c, don't store. Anyway, we don't care such low pT particles.
           continue;
         }
 
         int pdg = mctrack.pdgCode();
         if (
-          // abs(pdg) != 11 //electron // don't store all electron. This causes never-ending processing.
-          //&&
-          abs(pdg) != 22 // photon
+          abs(pdg) != 11                                               // electron
+          && (abs(pdg) != 22 || !IsPhysicalPrimary(mctrack, mcTracks)) // photon
           // light mesons
-          && (abs(pdg) != 111 || !mctrack.producedByGenerator()) // pi0
-          && (abs(pdg) != 113 || !mctrack.producedByGenerator()) // rho(770)
-          && (abs(pdg) != 211 || !mctrack.producedByGenerator()) // changed pion
-          && (abs(pdg) != 221 || !mctrack.producedByGenerator()) // eta
-          && (abs(pdg) != 223 || !mctrack.producedByGenerator()) // omega(782)
-          && (abs(pdg) != 331 || !mctrack.producedByGenerator()) // eta'(958)
-          && (abs(pdg) != 333 || !mctrack.producedByGenerator()) // phi(1020)
+          && (abs(pdg) != 111 || !IsPhysicalPrimary(mctrack, mcTracks)) // pi0
+          && (abs(pdg) != 113 || !IsPhysicalPrimary(mctrack, mcTracks)) // rho(770)
+          && (abs(pdg) != 211 || !IsPhysicalPrimary(mctrack, mcTracks)) // changed pion
+          && (abs(pdg) != 221 || !IsPhysicalPrimary(mctrack, mcTracks)) // eta
+          && (abs(pdg) != 223 || !IsPhysicalPrimary(mctrack, mcTracks)) // omega(782)
+          && (abs(pdg) != 331 || !IsPhysicalPrimary(mctrack, mcTracks)) // eta'(958)
+          && (abs(pdg) != 333 || !IsPhysicalPrimary(mctrack, mcTracks)) // phi(1020)
```

Comment:
the check` || !IsPhysicalPrimary(mctrack, mcTracks))` probably can be changed to only check this once right?

Reply:
probably, yes. I see your point. I will fix this in next commits.